dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -172,12 +182,20 @@ case class BinaryOutcomeDLCWithSelf(
   def createFundingTransaction: Future[Transaction] = {
     val output: TransactionOutput =
       TransactionOutput(totalInput, fundingSPK)
+    val localChange =
+      TransactionOutput(Satoshis(localFunding) - localInput, localChangeSPK)
+    val remoteChange =
+      TransactionOutput(Satoshis(remoteFunding) - remoteInput, remoteChangeSPK)
 
-    val outputs: Vector[TransactionOutput] = Vector(output)
+    val outputs: Vector[TransactionOutput] =
+      Vector(output, localChange, remoteChange)
     val txBuilderF: Future[BitcoinTxBuilder] =
-      BitcoinTxBuilder(outputs, fundingUtxos, feeRate, changeSPK, network)
+      BitcoinTxBuilder(outputs, fundingUtxos, feeRate, emptyChangeSPK, network)
```

Comment:
So I think i understand why this is needed, but it seems like the better way to handle this is with an `Option` type where the `TxBuilder` fails if there is change. If we have a bug where we send money to the empty changeSPK that means that money is lost forever. This is probably something else that should be addressed in #897 

One way I would maybe feel a bit more comfortable with this is if we use the [`sign()`](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala#L216) method that allows you specify an invariant about the resulting transaction. The invariant in this case would be that we do NOT have an output that is the empty SPK.

Reply:
Added the invariant for now and opened #961 for later